satpy/readers/slstr_l1b.py

```diff
@@ -250,13 +250,22 @@ def __init__(self, filename, filename_info, filetype_info):
         self._start_time = filename_info['start_time']
         self._end_time = filename_info['end_time']
 
-        cart_file = os.path.join(
+        carta_file = os.path.join(
+            os.path.dirname(self.filename), 'cartesian_a{}.nc'.format(self.view[0]))
+        self.carta = xr.open_dataset(carta_file,
+                                     decode_cf=True,
+                                     mask_and_scale=True,
+                                     chunks={'columns': CHUNK_SIZE,
+                                             'rows': CHUNK_SIZE})
+
```

Comment:
This could be made a separate function, as the code is reused a couple of lines down.

Reply:
Done!